src/pool/srv_pool.c

```diff
@@ -1195,7 +1195,7 @@ read_db_for_stepping_up(struct pool_svc *svc, struct pool_buf **map_buf,
 	ABT_rwlock_rdlock(svc->ps_lock);
 
 	/* Check the layout version. */
-	d_iov_set(&value, &global_version, sizeof(global_version));
+	d_iov_set(&value, &svc->ps_global_version, sizeof(svc->ps_global_version));
```

Comment:
ps_global_version need be bumped if 'pool upgrade' succeed, eg in ds_pool_mark_upgrade_completed().
Though now it will not cause problem, because newer version will use new format, but ps_global_version might be stale.

Reply:
> ps_global_version need be bumped if 'pool upgrade' succeed, eg in ds_pool_mark_upgrade_completed().
> Though now it will not cause problem, because newer version will use new format, but ps_global_version might be stale.

Right !! Thanks for pointing it @wangshilong !! I have pushed a new commit to fix this.